CraftsmenLtd/BloodConnect
# I-#4 basic config files added

Fixes #4 


## What kind of change does this PR introduce (check at least one)

- [x] Other (please specify): Tooling and configuration

## How would you describe the changes in a changelog (1-2 lines)?

Basic tools and configurations added.
- eslint
- jest
- tsconfig
- babel.config
- nvmrc

Build system is considered to be integrated at service level implementation. The basic setup will be done in the first service implementation PR. IAC and CI/CD related configurations have separate tickets defined.

## How to test
Add some basic `js` and `ts` modules in `core/application`/`commons`. 
- The IDE will show error if code does not conform with rules provided in `.eslintrc.js`.
- Import in modules with `@commons` and `@application`. Should not result in any error.
- Write some tests and execute them. The tests should execute successfully.

## Checklist


## Reviewer Request (optional)
Eslint rules will modify from time to time once actual code is written for the initial few months. In current implementation, the rules are added based on my own preferences. If there are other rules you would like to add, or if there are any other configuration files to be added, please mention as comment.

## Deploy risk & follow up
